services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/steps/SwitchSteps.java

```diff
@@ -72,6 +73,23 @@ public void selectARandomSwitch(String switchAlias) {
         topologyUnderTest.addAlias(switchAlias, theSwitch);
     }
 
+    @Given("^select a random switch with Openflow version '(.*)' and alias it as '(.*)'$")
+    public void selectARandomSwitchWithSpecificOfVersion(String ofVersion, String switchAlias) {
+        List<Switch> switches = getUnaliasedSwitches();
+        List<Switch> ofSwitches = new ArrayList<>();
+
+        for (Switch s: switches) {
+            if (ofVersion.equalsIgnoreCase(s.getOfVersion())) {
+                ofSwitches.add(s);
+            }
+        }
+
+        Random r = new Random();
+        Switch theSwitch = ofSwitches.get(r.nextInt(ofSwitches.size()));
```

Comment:
I have always thought that random input data is inacceptable for testing. Because it eliminate repeatability.

Reply:
That's true. This happens due to the fact that we do not hardcode our topology, because we are not creating the required topology from test (the physical one) and depend on config. This means that topology can be changed at any time and our tests will not be affected.
This, unfortunately, denies us from using concrete switch ids in tests, but, on the other hand, allows our tests to be run on any topology without changing/recompiling the code. In perfect scenario such methods should return the 'same' switch every time, so that test itself won't see the difference. Does it make sense?

Reply:
Agree with @surabujin. We can just order `switches` by switchId and pick the first one.

Reply:
Actually, randomness of switch selection allowed us for the eventual reliability - we were testing all switch makes over time. :)
However, you're totally correct about issues with the reproduceability, tests should definitely be deterministic. I have opened additional issue so that we can reach both goals simultaneously.